# Ejemplo Mardia (1979) page 127
<img src="https://raw.githubusercontent.com/fhernanb/fhernanb.github.io/master/my_docs/logo_unal_color.png" alt="drawing" width="200"/>

En este ejemplo se desea estudiar el siguiente conjunto de hipótesis usando un nivel de significancia de 0.05.

$H_0: \boldsymbol{\Sigma} = 
\begin{pmatrix}
100 & 0 \\
0 & 100
\end{pmatrix}
$

$H_A: \boldsymbol{\Sigma} \neq 
\begin{pmatrix}
100 & 0 \\
0 & 100
\end{pmatrix}
$

Suponiendo que la matriz de covarianzas muestral es

\begin{equation}
\mathbf{S} = 
\begin{pmatrix}
91.481 & 66.875 \\
66.875 & 96.775
\end{pmatrix}
\end{equation}

con $n=25$ sujetos que fueron los que participaron en el estudio.

In [1]:
import numpy as np
import scipy.stats

A continuación los datos del ejemplo

In [2]:
Sigma0 = np.matrix([[100, 0],
                    [0, 100]])
print(Sigma0)

print("\n")

S = np.matrix([[91.481, 66.875],
               [66.875, 96.775]])
print(S)

[[100   0]
 [  0 100]]


[[91.481 66.875]
 [66.875 96.775]]


## Prueba razón de verosimilitud de Mardia (1979) página 126 expresión 5.2.7

El estadístico está dado por

$$
-2\log(\lambda) = n \, trace(\mathbf{S} \, \boldsymbol{\Sigma}_0^{-1}) - n \, \log(|\mathbf{S} \, \boldsymbol{\Sigma}_0^{-1}|) - n \, p
$$

Cuando $H_0$ es verdadera, $-2\log(\lambda) \sim \chi^2_{p(p+1)/2}$.

In [3]:
n = 25
p = 2

matriz_aux = np.matmul(S, Sigma0.I)
print(matriz_aux)

[[0.91481 0.66875]
 [0.66875 0.96775]]


In [4]:
minus_2_log_lambda = n * np.trace(matriz_aux) - n * np.log(np.linalg.det(matriz_aux)) - n * p
print(minus_2_log_lambda)

17.69779691232469


Para calcular el valor crítico con un $\alpha=0.05$ hacemos lo siguiente

In [5]:
scipy.stats.chi2.ppf(q=1-0.05, df=p*(p+1)/2)  # Valor crítico

7.814727903251179

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $-2\log(\lambda)=17.70$, que es mayor que el valor crítico $7.81$, entonces si hay evidencias suficientes para rechazar $H_0: \boldsymbol{\Sigma} = \boldsymbol{\Sigma}_0$.
</div>